Sentinel Redesigned

lKiseki opened the pull request:
# About The Pull Request
Redesigns the Sentinel caste. Pushes it into a rushdown + debuff management niche, centered around quick assaults and managing the new Intoxicated debuff to reap benefits.

## Additions
### Debuff: Intoxicated
Most of the Sentinel's abilities now inflict the Intoxicated debuff, which will deal damage over time.
- Deals damage over time equal to 1.0 base damage + 0.1 per stack, for a minimum of 1.0 damage and a maximum of 3.0 damage. This damage is applied every 2 seconds.
- Can be stacked.
  - Each stack extends the duration by 2 seconds, up to a maximum of 30 stacks or 60 seconds.
  - At 20 stacks, inflicts 1 stack of stagger and slowdown.
- An icon will appear under a victim, indicating the level of Intoxication. A blinking icon means it has stacked to a critical level, which is important for the **Drain Sting** ability featured below. The critical level is 20 stacks.
![debuffcounter](https://user-images.githubusercontent.com/59634950/206897665-f0ad7920-4a4e-4fd0-b6c1-8a7b8227f396.gif)
- Counts for the Defiler's Defile ability, increasing its final damage. Using Defile in this manner will decrease the amount of stacks by 10.
- Sanguinal will also build up 3 stacks of the Intoxicated debuff for every tick it is in a victim's system.
- Can be cleared by Resisting. Using Resist in this manner will reduce the current amount of stacks by 10% _(see: 3 stacks)_, but requires a 3-second channel.
- The Mimir module and the Ryetalyn chemical will grant resistance to this debuff, making it decay twice as fast.

### Toxic Spit
Spit acid at your victim, inflicting the Intoxicated debuff.
- Has a plasma cost of 30.
- Deals 12/14/15/16 burn damage, based on maturity.
- Inflicts 2/3/4/5 stacks of the Intoxicated debuff, based on maturity.
- Goes past allied xenos.
![toxicspit](https://user-images.githubusercontent.com/59634950/206835301-25520921-d6d9-4c73-b3e5-ce1fb78defce.gif)

### Toxic Slash
Imbue your claws with acid, inflicting the Intoxicated debuff with every slash.
- Has a plasma cost of 100, and a cooldown of 10 seconds.
- Buff ends after 4 seconds, or after 3 slashes have been dealt.
- Each slash will inflict 2/3/4/5 stacks of the Intoxicated debuff, based on maturity.
![toxicslash](https://user-images.githubusercontent.com/59634950/206835372-476d9615-9f3b-49b9-9748-51c7e4301476.gif)

### Drain Sting
Sting your victim, draining some Intoxicated stacks and gaining benefits from it.
- Has a plasma cost of 75, and a cooldown of 25 seconds.
- Removes 30% of the current stack amount on use.
- If the victim is above 20 stacks, this ability will increase all armor ratings by 20, for 10 seconds.
- Inflicts knockdown. The duration of this knockdown is proportional to the amount of stacks the victim has, minus 1 second. For example; 30 stacks equals 2 seconds, and 15 stacks equals 0.5 seconds. The minimum duration is 0.1 seconds.
- Deals a proportional amount of damage. The formula for this damage is: `( Amount of stacks * 6 ) / 5`. This translates to a minimum of 1.2 damage and a maximum of 36 damage.
- Heals a proportional amount of health. The formula for this heal is: `Amount of stacks * 6`. This translates to a minimum of 6 health and a maximum of 180 health.
- Restores a proportional amount of plasma. The formula for this restoration is: `( Amount of stacks * 6 ) * 3.5`. This translates to a minimum of 21 plasma and a maximum of 630 plasma.
![drainsting](https://user-images.githubusercontent.com/59634950/206835707-9d15b874-bcdf-4889-b93e-080951af153b.gif)

### Primordial: Toxic Grenade
Throw a ball of resin, which contains a toxin. This ball of resin will detonate shortly after landing, spreading a gas that inflicts the Intoxicated debuff to anyone caught inside its range.
- Has a plasma cost of 200, and a cooldown of 50 seconds.
- Deals 20 burn damage _(before reductions)_ for every tick spent inside it.
- Inflicts 10 stacks of the Intoxicated debuff for every tick spent inside it.
- Allows other gases through, allowing it to synergize with other xenos such as the Defiler.
![ezgif com-gif-maker(29)](https://user-images.githubusercontent.com/59634950/206835915-db3790f7-cde2-4be1-86a9-7962c56d4e82.gif)

## Changes / Removals
### Acid / Neurotoxin Spit
These two spit options have been removed from the Sentinel. The Toxic Spit fulfills the Acid Spit's functions in a better way, and Neurotoxin isn't particularly something I'd like to keep in this redesign.

### Neurotoxin Sting
This ability has been removed. As previously mentioned, Neurotoxin has no place in this redesign.

### Caste Stats
- Melee damage has been reduced from 16/16/17/17 to 14/14/16/16 _(depending on maturity)_. The addition of the Intoxicated debuff has given the Sentinel a considerable boost in DPS, and I'd rather the caste be focused on that debuff rather than in raw damage.
- Movement speed has been increased from -0.8 to -0.9 _(the higher the value, the faster)_. Since the Sentinel is now required to enter melee range to make the most of its kit, I felt a movement boost was necessary, but didn't want to take it too far either.

# Why It's Good For The Game
The Sentinel has been in a terrible spot for the longest time, and has been regarded as a lesser Spitter. This PR aims to amend that by giving the Sentinel a more unique identity and pushing it into the more enjoyable niche of rushdown + debuff management, which in turn adds more depth to the caste.

# Changelog
:cl: Lewdcifer
add: The Sentinel caste has been redesigned, and is now centered around managing the Intoxicated debuff, which will give victims perjudicial effects.
imageadd: Victims affected by this debuff will display an icon on any given xenomorph's HUD, indicating just how much of that debuff they have.
add: New Sentinel ability: Toxic Spit. Spit at a target enemy to inflict them with the Intoxicated debuff.
add: New Sentinel ability: Toxic Slash. Imbue your claws with toxins, inflicting the Intoxicated debuff.
add: New Sentinel ability: Drain Sting. Sting your victim, draining their Intoxication. This will grant you several benefits if used on a victim with high Intoxicated stacks.
add: New Sentinel ability: Toxic Grenade. Throw a ball of resin which will burst into a large gas cloud. Victims caught inside its range will be inflicted with the Intoxicated debuff, and take considerable burn damage.
del: The Sentinel no longer has access to acid and neurotoxin spits.
del: The Sentinel no longer has access to the Neurotoxin Sting ability.
del: Removed the Neurogas Grenade.
balance: Sentinel slash damage reduced from 16/16/17/17 to 14/14/16/16.
balance: Sentinel movement speed increased from -0.8 to -0.9 (the higher the value, the faster).
/:cl:

Waselon commented:
Intoxicated debuff or drain sting should deal more damage with stacks imo

lKiseki commented:
> Intoxicated debuff or drain sting should deal more damage with stacks imo something like 0.8 base damage + 0.1 damage per stack

Might not be such a bad idea to do this. 

> Intoxicated debuff or drain sting should deal more damage with stacks imo something like 0.8 base damage + 0.1 damage per stack

Not a bad idea, admittedly. I'll give this some consideration.

Lumipharon commented:
Cool idea

Waselon commented:
Maybe spit should inject some toxin, so Defiler can take advantage of this

CabbageOnTable commented:
I actually think that is no point to remove neuro spit. You'd like given no escape ability and bad damage to melee caste now. Maybe remove movement speed buff and return neuro spit?

maikilangiolo commented:
> * Melee damage has been reduced from 16/16/17/17 to 14/14/16/16 _(depending on maturity)_. The addition of the Intoxicated debuff has given the Sentinel a considerable boost in DPS

It's... literally 1 less damage. Either don't nerf it or actually nerf it.

SpiderPsycho commented:
Getting rid of neurotoxin is one thing but the acid damage being proposed is worse than current sentinel spit. This is a shadow-nerf and a half if I've ever seen one.

Waselon commented:
Toxic Spit - needs more damage (current acid spit deals 30 damage)
Drain Sting - needs more damage
Intoxicated debuff - needs to apply slow, stagger and stamina damage, needs to deal more DoT damage
If you are forcing Sentinel into melee range, give it more health/armor

KerbalCitizen commented:
So 10 spits or 10 toxic slashes to hit max intoxication stacks, then you do your epic combo finisher: drain sting for whopping 1 second stun and 24 (dunno type of) damage. Am I alone in thinking that drain sting is just horrible?

Toxic spit does less damage than current sent acid spit, which is already garbage. 

The 0.8 damage over time from being intoxicated can easily be outhealed assuming it's brute or burn. I guess you could argue that it makes marines heal a bit slower, but corpsman just kits their wounds and it's irrelevant at that point. 

From my understanding of this PR, current sent seems way stronger than this new redesigned one. Other castes like defender just seem so much better. Defender tail sweep is basically redesigned sent's drain sting without needing prep (10 toxic spits/ toxic slashes), but defender also has forward charge for long stun or quick retreat/ closing distance. Has crest defense to survive more when speed isnt important, fortify to laugh at explosions, bodyblock, or just surviving more bullets.

srxtreme commented:
For purposes of changes, I'll consider acid spit changed to toxic spit, neurotoxic spit removed (not improved), neurotoxin sting removed.
Neurotoxin sting removed = while leaving someone with 40u of neurotoxin is funny, neuro sting was far from the most important or crucial ability in sentinel's kit. Eh, could be worse
Neurotoxic spit removed = no more stam damage, no more slowdown, no more stagger, no more keeping an important/dangerous marine locked down for a good period of time. Kills sentinel's duelling potential due to lack of stagger debuffing marine damage, kills sentinel's potential to defend themselves and others from rushers due to no slowdown, stagger, or knocking marines out of sprint, kills sentinel's potential to combo with defiler or other neuro/stam damage dealers such as Prae Boiler Defiler Carrier Sent, kills sentinel's ability to keep important targets staggered or forced out of combat due to stamina loss. Last but not the least kills Sentinel's identity as the neuro guy. Why would someone do this I don't know, but it's probably not relevant right now. Awful change, the lone saving grace is that you can no longer use up all your plasma on neuro spit (since it's gone, get it).

Acid spit damage reduced from 18 (9 post mitigation) to 14 (7 posti mitigation), now applies 2 stacks of armor-bypassing effect that deals (0.6? 0.8? 2?) burn damage per 2 seconds (roughly one life tick, which is the similar speed chems metabolize at). For comparison taking kelotane + tricordrazine heals 2.3 burn damage per two seconds. Without neurotoxin spit, acid is sentinel's only option for ranged damage. This makes sentinel a shitty acid spitslasher. 

Drain sting removes the Intoxicated stacks to deal up to (24?) burn damage, while simultaneously healing and buffing sentinel. This is alright, and the only actually good thing in the entire rework.

In summary: kills sentinel's identity, turns it into a shitty spitter which has to go into melee to land a finisher presumably. While having no stuns, no debuffs, no survivabilty bonuses (until it lands the finisher), and 0 CC to actually get there.

Giving sentinel access to choice between toxin and neuro spit (or a modified neuro spit), or adding stagger + slow to toxin spit should allow sentinel to remain a solid duelist (if that's what sentinel's supposed to do anyways), while still retaining its old identity to an acceptable degree.

helper-2 commented:
That makes sents completely useless against robots (as if they aren't weak already). Basically, 8 melee damage, and no range damage against robot with 150 hp, pretty much forces you to avoid them.

TiviPlus reviewed (approved):
Only complaint is you could consider making particle count scale with the amount of stacks you have

Naaanii commented:
I'd say keep all neuro abilities till sentinel actually gets health and defensive stat buffs and let people experiment with the now 3 types of abilities.

SpiderPsycho commented:
I can't wait till you make all burn damage even more useless by making it DoT.

Jordanstolethepreciousslam commented:
What the fuсk.

-Deleted stagger and slow from projectile(The only thing that made sentinel slightly stronger than larva)
-Weak ass damage on hit(4 less than worst acid spit in the game does)
-Pathetic ass damage over time(Outhealed by trico.  considerable boost in DPS my ass)
-No stamina damage on hit wtf lmao why even bother
-Killed primo by making gas transparent. For a super strong damage that is outhealed by trico no less.
-The super powerful combo requires 7+ spits and going into melee(no stagger reducing marine damage remember) deals whopping 25 damage. 

This is some MJP level garbage.

MiniMeatwad commented:
Where's the stagger and stamina drain? Although this seems quite better than how neuro worked as a chem, it should still give stagger and stamloss as its primary effect. Sentinel wasn't about DPS, it was about support.

Pariah919 commented:
This seems unnecessary considering Sentinel is fine design wise. And overall looks like a pretty bad nerf to the caste, removes support castes from the xeno roster and shifts it to a fragging role which is poor for the caste design. I get what you're going for but I think Sentinel is fine as is, and giving it this with Neuro (And assuming acid goes in favor of this instead) untouched is too far of a buff.

Edelor68 commented:
I don't really see the intended use for the Toxic Slash if each slash applies the same amount of intoxication as the spit.
It is way riskier to go into melee range and costs double the amount of just spitting on someone 3 times ( 150 vs 76 )
I don't see a situation where you should ever use this instead of spitting.

Losenis commented:
Ghetto defiler for those that had the T3 nabbed.I like it.

DeltaFire15 commented:
Removing sent's support potential towards neutralizing fast marines (neuro's stagger-stam) seems like a terrible thing, considering it is the only sub T3 ranged slow (well except hivelord but those tend to have issues with aggressively using it) and is very fun to play around with.
In exchange, they get:
Low damage DoT that needs many stacks where the marine can just, disengage and the sent can do nothing against.
Finisher with very low rewards for massive risk (going into melee as the squishiest ranged caste).
Primordial that just applies some of the low-dot-tox and doesn't even block LOS as opposed to the gas grenade base.
In addition, all potential for dealing with robots (the weal-ish acid spit) also got removed so I guess if you see a robot you get to just cry.

All in all, seems mainly like a sent nerf.

Losenis commented:
Oh yeah robots do feel like they'll be a very big problem in principle

Redrover1760 commented:
> For purposes of changes, I'll consider acid spit changed to toxic spit, neurotoxic spit removed (not improved), neurotoxin sting removed. Neurotoxin sting removed = while leaving someone with 40u of neurotoxin is funny, neuro sting was far from the most important or crucial ability in sentinel's kit. Eh, could be worse
> 
> Neurotoxic spit removed = no more stam damage, no more slowdown, no more stagger, no more keeping an important/dangerous marine locked down for a good period of time. Kills sentinel's duelling potential due to lack of stagger debuffing marine damage, kills sentinel's potential to defend themselves and others from rushers due to no slowdown, stagger, or knocking marines out of sprint, kills sentinel's potential to combo with defiler or other neuro/stam damage dealers such as Prae Boiler Defiler Carrier Sent, kills sentinel's ability to keep important targets staggered or forced out of combat due to stamina loss. Last but not the least kills Sentinel's identity as the dedicated neurotoxin spitting xenomorph. Why would someone do this I don't know, but it's probably not relevant right now. Awful change, the lone saving grace is that you can no longer use up all your plasma on neuro spit (since it's gone, get it).
> 
> Acid spit damage reduced from 18 (9 post mitigation) to 14 (7 posti mitigation), now applies 2 stacks of armor-bypassing effect that deals (0.6? 0.8? 2?) burn damage per 2 seconds (roughly one life tick, which is the similar speed chems metabolize at). For comparison taking kelotane + tricordrazine heals 2.3 burn damage per two seconds. Without neurotoxin spit, acid is sentinel's only option for ranged damage. This makes sentinel a shitty acid spitslasher.
> 
> Drain sting removes the Intoxicated stacks to deal up to (24?) burn damage, while simultaneously healing and buffing sentinel. This is alright, and the only actually good thing in the entire rework.
> 
> In summary: kills sentinel's identity, turns it into a shitty spitter which has to go into melee to land a finisher presumably. While having no stuns, no debuffs, no survivabilty bonuses (until it lands the finisher), and 0 CC to actually get there.
> 
> Giving sentinel access to choice between toxin and neuro spit (or a modified neuro spit), or adding stagger + slow to toxin spit should allow sentinel to remain a solid duelist (if that's what sentinel's supposed to do anyways), while still retaining its old identity to an acceptable degree.

I like sentinel as it is right now and agree with this. Sure, it may need some buffs or something, but neurotoxin spit is really good utility for any hive and quite fun to use, well still not being too strong at all, and it'd be a shame to see it go.

ThelastArchiver commented:
removing sentinels neurotoxin spit is a horrible idea without the slowdown and the funny chem purging sentinels wont be able to punish premedders or maybe help slow down a push with their gas and spits

Goblinu commented:
I have to agree with what most people are saying here, this kills spitter identity. What youve done is essentially removed sentinel as the dedicated neurotoxin dispenser and instead made it a worse defiler, why would anyone play sentinel when defiler is more fun and engaging? 

If anything sentinel should be redesigned to keep its ranged debuff support identity, but this is just "deal damage but in a funny way" thats something the runner. This rework, in my humble opinion makes it even less attractive to pick sentinel and just gun for spitter that has a AoE knockdown and stun instead of a weak ass knockdown that requires a ton of buildup

Lumipharon commented:
After thinking about it, while  I do think that the idea is cool, current sentinel fills a useful support niche via severe Stam damage and stagger, while this is just another damage dealing caste with no support utility.
 It also probably makes it more of a 'worse spitter' in the sense that spitters also do direct damage, but do more up front, while also having more utility.

On the other hand, I think this would be cool as a new caste if you could get a sprite.

shelteredfreak commented:
This feels like Defiler lite, and I don't think Sentinel is regarded as a stepping stone to spitter, someone has clearly been spoonfeeding you info.

I'm not saying I disapprove or hate the design, I'm just saying that it feels slightly unwarranted. Sentinel is enjoyable as a lighter and flightier Prae without pheros. Although it has weaker neuro spit, neuro is its main feature. It's not entirely devoid of identity, I personally enjoy my time using neuro sting and neuro spit to slow and stagger marines into performing less effective. This debuff also kinda removes a lot of synergy that it had with Defiler, which is greatly disappointing. 

I don't hate it, and I'm not against it. But it feels unneeded. If anything were to be done to Sentinel, I would say improve its plasma. If you want to buff it, buff the spit rate, buff the plasma capacity, buff the plasma regen. It's an effective caste at what it does **if used right**. Just because someone tells you a caste is bad for this and that reason, does not mean it is bad, it just means they think its bad.

Most castes can be used effectively if you put a few hours into the caste to learn it, binding it and learning its abilities. So what I'm saying is: Every caste has a use, not everything is just a better x. Or a stepping stone from y to x. Some things are best left untouched.

But I am going to restate this for the third time so I am very clear. **I do not hate this, and I do not want it denied. I just want some more thought put into it before it gets merged at all.**

ilikesaying commented:
![image](https://user-images.githubusercontent.com/65475697/206892319-4da2a064-868d-4252-95f4-0c747e7d7b54.png)
isn't 2 x 20 = 40
therefore it's a maximum of 40 seconds, not 3 minutes? or am i missing something?

Aporhtonoma commented:
This feels like a sentinel nerf to be honest old senti had neuro going for it making it able to synergize with other xenos and be best buds with a defiler.Sure it wasnt the toughest or most oppresive caste but it had its use and fitted the xeno caste team.
But straight up removing neuro and giving it a debuff instead makes it feel like a lone wolf xeno that does its own thing.No stacking neuro with others,no more epic frens with defilers.Hell not even taking a chem slot from the rine making them dizzy and dot damage can be outright ignored by using trico.If it was a new caste or you made the debuff a xeno chem instead sure i could see it work a bit more but currently you are just making a ranged squishy caste with no dodge into a close to medium range squishy caste with no dodge and synergy among itself.
TLDR feels like a nerf, effort is there,intent is there but still feels like nerf

lKiseki commented:
Gonna go ahead and address several concerns from this PR.

Addressing several concerns regarding the removal of Neurotoxin and its relevancy as an important support option; it is my belief that Neurotoxin was quite powercrept and unfun to deal with.
Continuous stagger and slowdown, on top of stamina damage which - with enough of this damage type - would also translate to _more_ stagger and slowdown. There were many mentions of several ways to counterplay this sort of thing _(see: water, dylovene, etc.)_, but even so, that was just not fun to deal with.
This debuff aims to change that by providing a less crippling option while giving Sentinel a considerable boost in DPS, which pushes it out of its support role to a degree, but does compensate it by giving it a glass cannon / bruiser type of role.
##
I am going to emphasize the following bit of text because people should notice it.
### If people preferred Sentinel as a support role, I can weigh some options and consider shifting Sentinel from a DPS centered role back to a support role by changing some of the damage types that Sentinel deals with, from burn to stamina. However, I do believe that Neurotoxin was overloaded with stagger and slowdown, something which I would not do myself. Dealing purely with burn and stamina seems like a more controlled way to handle this, in comparison to before.
The above would most likely mean I'd have to heavily alter this redesign and tweak several values for the purposes of balance, meaning it's a big change. I would appreciate some thoughts on this little bit here before I commit to it.
##
In any case, I got a great deal of good feedback which I will be accepting. **All of these changes will reflect in the next commit, which I am holding back on so I can keep the Test Merge Candidate tag in this PR**.

@Waselon offered some good suggestions, and I'll be implementing some of them.

> Intoxicated debuff or drain sting should deal more damage with stacks imo something like 0.8 base damage + 0.1 damage per stack

- The debuff will now deal damage equal to 1.0 base damage + 0.1 per stack, for a minimum of 1.0 damage and a maximum of 3.5 damage. I understand that many people were complaining about the damage being too low and easily outhealed by chems like Kelotane and Tricordrazine, hence the increased damage.
- To compensate for the above, and as a means to provide more counterplay to this debuff, the status effect can now be dealt with by Resisting. This will reduce the current amount of stacks by 20%, but will immobilize you for 5 seconds, preventing you from doing anything.

> Maybe spit should inject some toxin, so Defiler can take advantage of this

- The debuff will now count towards the Defiler's Defile sting, increasing its final damage but reducing the current amount of stacks by 10.
- Sanguinal will also build up 2 stacks of the debuff every tick.

> Toxic Spit - needs more damage

- Toxic Spit damage increased from 10/12/13/14 to 12/14/15/16 _(based on maturity)_.

> Drain Sting - needs more damage

- Drain Sting has had its damage increased, for a minimum of 1.5 damage and a maximum of 30 damage. Additionally, and in response to several concerns, I have made it so that Drain Sting will now inflict 5 stacks of stagger and slowdown, and apply a buff that will increase the Sentinel's soft armor ratings by 20. This buff lasts for 10 seconds.

##
@novaepee wrote the following:
> Is there an indicator for how many stacks a marine have for less than 15 stacks, the critical amount?
- The indicator itself is the bubble icon. The bubble icon is there if the victim has Intoxicated debuff, and will also start blinking if they have more than 15 stacks.

> If not, splitting intoxicated hub into 25%, 50%, and 75% before critical would be great information for sent players to have. If space permits it, a counter for intoxicated hub would be really great info OR you replace the less than 15 stacks into green numbers.

- I'm too lazy to make icon variants based on stack amounts, so I've decided to add a numeric counter on top of the bubble. This numeric counter will reflect the precise amount of stacks a victim has on them currently.
![debuffcounter](https://user-images.githubusercontent.com/59634950/206892949-e6eafd5d-1cf1-42ae-aa7c-5e80da0bb7ee.gif)

> Will intoxicated show up in the analyzer? It sounds like the counter to removing intoxicated is to walk it off.

- The debuff will not show up in the analyzer. I believe the presence of particles and the message logs make the presence of this debuff obvious enough, and with the addition of more counterplay, this is also relatively unnecessary now. If someone really wants it in the analyzer, though, they can add it later down the line.

##
@DeltaFire15 wrote the following:

> Low damage DoT that needs many stacks where the marine can just, disengage and the sent can do nothing against. Finisher with very low rewards for massive risk (going into melee as the squishiest ranged caste).

- As previously mentioned, the DoT has been buffed in accordance to some of the feedback I was given. However, I do understand that marines can just disengage and it's good bye stacks. To compensate for this and other concerns of a similar vein, not to mention the addition of more counterplay, I have increased the amount of stacks dealt by each of the Sentinel's abilities.
  - Each toxic spit will now inflict 3 stacks _(from 2 per spit)_.
  - Each toxic slash will now inflict 4 stacks _(from 3 per slash)_.

> Primordial that just applies some of the low-dot-tox and doesn't even block LOS as opposed to the gas grenade base.

- This does seem like a bad change on my part. I have made it so the gas will block LOS once again. Additionally, I noticed that it did not allow other xeno gases through, which I have changed. It should work similarly to Defiler gases in this regard, for reference.
- The gas will now deal 20 burn damage _(before armor reductions)_ to anyone standing inside it, and will also deal 5 stacks of Intoxicated per tick _(from 4 per tick)_.

> Oh and it also removes the ability to shoot past allies (like with neuro) to actually support an ongoing fight, which was a very good ability to have being as squishy as you are.

- Toxic Spit will now go through other xenos.

##
@ilikesaying wrote the following:

> ![image](https://user-images.githubusercontent.com/65475697/206892319-4da2a064-868d-4252-95f4-0c747e7d7b54.png) isn't 2 x 20 = 40 therefore it's a maximum of 40 seconds, not 3 minutes? or am i missing something?

Whoops, my bad. Yeah, it's 40 seconds, I just get confused between ticks and seconds since 20 ticks is 2 seconds. I was doing 20 x 20 _(which would be 400 ticks, too, meaning I'm wrong either way)_. I have corrected that bit of information to reflect the correct duration.

lKiseki commented:
wow rude i just made a comment but okay

SpiderPsycho commented:
Sentinel is literally the support T1 caste next to the dedicated builder/healer. Turning it into a DPS is silly.

This aversion to making xenos stronger than a marine on a 1-to-1 level is completely counterintuitive. The game is asymetric but for some reason developer logic has decided that the more numerous team should be stronger on a one to one basis than the team with less individuals. I say some reason but it's pretty obvious that there's a bias towards marines.

lKiseki commented:
> After thinking about it, while I do think that the idea is cool, current sentinel fills a useful support niche via severe Stam damage and stagger, while this is just another damage dealing caste with no support utility.

To answer this, I do believe there's some merit to a _support DPS_, since the DoT would quickly add up to any other damage source from other xenos, not to mention the damage that you yourself can deal alongside others. Just because you're not inflicting stamina damage and 2 other debuffs doesn't mean you're not contributing.

> It also probably makes it more of a 'worse spitter' in the sense that spitters also do direct damage, but do more up front, while also having more utility. The nature of the debuff also means there's an incentive to spam sentinels to stack debuff super fast, with no direct way to counter it, which might not be great from a gameplay/design point of view

To be very fair, stacking any caste in general tends to do this. It's not really one of my immediate concerns, admittedly.

lKiseki commented:
> Sprite should probably be updated to reflect the new role. Just a suggestion.

This is outside of my budget of zero (0) dollars. I apologize.

Al-1ce commented:
> This is outside of my budget of zero (0) dollars. I apologize.

Understandable. Didn't know people here actually buy sprites.

lKiseki commented:
> Understandable. Didn't know people here actually buy sprites.

Yep. Sadly, we don't have a dedicated xeno spriter or someone that's willing to do that for free _(unless some spriter really just likes something)_. Generally, for new xeno sprites, we have to actually pay for them. It was the case with Gorger and Warlock, and it'd also be the case with Sentinel if it were to happen.

lKiseki commented:
> Only complaint is you could consider making particle count scale with the amount of stacks you have

Donezo in the next commit.

carlarctg commented:
this would be great as a strain but yall not ready for that conversation

lKiseki commented:
> this would be great as a strain but yall not ready for that conversation

![pensivecowboy](https://user-images.githubusercontent.com/59634950/206899852-fae8a730-9085-4b91-a98e-861c567d5f8b.png)

Naaanii commented:
> ![image](https://user-images.githubusercontent.com/65475697/206892319-4da2a064-868d-4252-95f4-0c747e7d7b54.png) isn't 2 x 20 = 40 therefore it's a maximum of 40 seconds, not 3 minutes? or am i missing something?

Stacks cap at 20.

MiniMeatwad commented:
Most of the "OPness" from neuro spit was just the neuro chemical, which can go away because it's chem gameplay and it's bad. The immediate stamina damage and stagger with IFF helps prevent a marine from blasting Xenos into instant death with juggle DB.

When I talk about sentinel as a debuff caste, I don't mean it as an "apply lasting effect" caste, I mean its immediate effect of stagger, and 50% less damage output. 

When people talk about getting tons of kills with sentinel directly, it's because of the neuro chem which actually stuns people constantly as soon as it's in their system, on top of staggering them constantly (afaik), and giving endless slowdown.

Removing neuro chem is great for sentinel, but do leave an option to IFF spit a stagger projectile like sticky resin spit, but without the sticky resin.

DeltaFire15 commented:
>When people talk about getting tons of kills with sentinel directly, it's because of the neuro chem which actually stuns people constantly as soon as it's in their system, on top of staggering them constantly (afaik), and giving endless slowdown.

It does very much not do that.
What it does is dealing stam DoT scaling over how many cycles it has been in system, which turns into tox/oxy damage when you run out of stamina. Nothing about the reagent itself applies stagger or slow, unless you count draining your stamina as a indirect slow.
It does start causing microknockdowns when you are completetely out of stam (mostly when actively being spit at by dram-draining spit), but that isn't exactly an "all the time" either.

MiniMeatwad commented:
> It does very much not do that. What it does is dealing stam DoT scaling over how many cycles it has been in system, which turns into tox/oxy damage when you run out of stamina. Nothing about the reagent itself applies stagger or slow, unless you count draining your stamina as a indirect slow. It does start causing microknockdowns when you are completetely out of stam (mostly when actively being spit at by dram-draining spit), but that isn't exactly an "all the time" either.

Cool. I never looked into how the neuro chem worked, because it always felt funky, and like it was going to be removed eventually. Most of the kills you get as sentinel are from the micro stuns that people get stuck with when they have neuro.

Redrover1760 commented:
Honestly, I change my mind. I think this is a very good idea for a change. Sure, maybe it'd be a good idea to give Sentinel a support spit of some kind, but this kit is a lot more fun than current Sentinel, now that the numbers aren't undertuned, anyways.

Also, I don't think multiple sentinels teaming up on someone is a problem at all, as not only should Xenos should just be a lot stronger working together, but also that assumes other marines aren't there to help deal with the sentinels trying to do something... Which means they are already outnumbered anyways. (Not to mention, I really like the idea of sentinel squad, haha.)

novaepee commented:
> Yep. Sadly, we don't have a dedicated xeno spriter or someone that's willing to do that for free _(unless some spriter really just likes something)_. Generally, for new xeno sprites, we have to actually pay for them. It was the case with Gorger and Warlock, and it'd also be the case with Sentinel if it were to happen.

I'm not paying for another xeno sprite

lKiseki commented:
Made a few changes based on overall feedback and the live test.
##
### Bugs + Oversights
- Fixed a bug where the Defiler's Defile ability would not work correctly in the absence of the Intoxicated debuff, due to a missing check.
- Fixed a bug where the Toxic Grenade gas was not inflicting damage in certain scenarios.
- Fixed an oversight where the Toxic Grenade gas would inflict 3 stacks of stagger and slowdown each.
##
### Debuff: Intoxication
- Maximum amount of stacks possible increased to 30 _(from 20)_.
- Damage over time adjusted to account for the above change. Remains the same _(minimum of 1.0 and maximum of 3.0)_.
- Will now inflict 1 slowdown stack when at 10 stacks or more.
- Will now inflict 1 stagger stack when at 20 stacks or more.
- Certain things will now allow you to be **resistant** to the debuff. This limits the maximum amount of stacks that you can have _(currently reduces it to 15, from 30)_, which in turn affects many of the Sentinel's abilities. If you were already over the new limit, the stacks will be adjusted to that limit.
  - The Mimir modules will now grant resistance to this debuff.
  - Ryetalyn will now provide resistance to this debuff.
  - Resistance may be added to more things if this is deemed a good addition as per further live testing.
- Similarly, the option to be **immune** to the debuff exists in code, but is currently unused. This prevents you from getting this debuff. Some of the Sentinel's abilities will change to reflect this immunity, as well.
##
### Toxic Slash
- Removed the species check, making it usable on Combat Robots and Synthetics.
- Can now be used while buckled, or while riding a Crusher.
- Plasma cost reduced to 100 _(from 150)_.
- Will now inflict 5 stacks per hit _(from 4)_.
##
### Toxic Spit
- Plasma cost increased to 30 _(from 28)_.
- Will now inflict 5 stacks per hit _(from 3)_.
##
### Drain Sting
- Removed the species check, making it usable on Combat Robots and Synthetics.
- Will no longer inflict Stagger and Slowdown stacks.
- The knockdown effect is now always applied.
- The knockdown duration has been adjusted to be proportional to the amount of stacks the victim has. 30 stacks translates to 3 seconds, 15 stacks translates to 1.5 seconds, and so on.
- The maximum amount of health restoration is now 180 _(from 120)_.
- Damage formula changed to `( Stacks * 6 ) / 5`. This now translates to a minimum of 1.2 damage and a maximum of 36 damage. The previous formula was `( Stacks * 6 ) / 4`, which translated to a minimum of 1.5 damage and a maximum of 30 damage.
- Plasma gain formula changed to `( Stacks * 6 ) * 3.5`. This now translates to a minimum of 21 and a maximum of 630. The previous formula was `( Stacks * 6 ) * 3`, which translated to a minimum of 18 and a maximum of 360.
- Will no longer remove all stacks of Intoxication. Instead, it will remove 30% of the current amount of stacks _(30% of 30 is 9, 30% of 15 is 4.5, and so on)_.
##
### Toxic Grenade
- Plasma cost reduced to 200 _(from 250)_.
- Range radius increased to 4 _(from 3)_.
- Gas duration reduced to 2 seconds _(from 6)_.
- Toxic Grenade gas will now inflict 10 stacks for every tick spent inside it _(from 5)_.
- Toxic Grenade gas will now deal 20 damage _(before reductions)_ for every tick spent inside it _(from 18)_.
##
### Defiler: Defile
- Due to the maximum amount of stacks increase for Intoxication, the damage bonus for Defile has been increased as a result.
- Will no longer remove all stacks of Intoxication. Instead, it will halve the current amount of stacks.
##
### To-Do
- Major visual redesigning. I am currently very unhappy with how visually boring some of these abilities look, so I'll be looking into making them much more satisfying from a sensory standpoint.
- Test and experiment these balance changes. Balance is hard!

MiniMeatwad commented:
Damn, will have to play hivelord to use stagger spit now.

MiniMeatwad commented:
Damn, will have to play hivelord to use stagger spit now. Aren't T2 slots infinite now, though? Or was that a bug?

Lumipharon commented on code/modules/mob/living/living_defense.dm:
```diff
@@ -246,6 +246,13 @@
 		adjustFireLoss(S.strength * rand(20, 23) * protection)
+	if(CHECK_BITFIELD(S.smoke_traits, SMOKE_XENO_TOXIC))
+		if(!HAS_TRAIT(src, TRAIT_INTOXICATION_IMMUNE))
+			if(has_status_effect(STATUS_EFFECT_INTOXICATED))
```
Ditto on what I said on the projectile.

Lumipharon commented on code/modules/mob/living/carbon/xenomorph/castes/sentinel/abilities_sentinel.dm:
```diff
@@ -1,36 +1,220 @@
-	name = "Throw neurogas grenade"
-	action_icon_state = "gas mine"
-	mechanics_text = "Throws a gas emitting grenade at your enemies."
+/datum/ammo/xeno/acid/toxic_spit
```
Shouldn't all this ammo code be in the ammo datum file?

Lumipharon commented:
Also just in terms of walance numbers, hard stun for up to 3 seconds seems rather strong, on top of the fat heal and other buffs. This seems like a bit of a 1v1 monster if played well.

Naaanii commented:
> Damn, will have to play hivelord to use stagger spit now. Aren't T2 slots infinite now, though? Or was that a bug?

Bug that was fixed.

lKiseki commented on code/datums/status_effects/debuffs.dm:
```diff
@@ -418,3 +418,62 @@
+		stacks = max_stacks
+	var/debuff_damage = SENTINEL_INTOXICATED_BASE_DAMAGE + round(stacks / 10)
+	debuff_owner.adjustFireLoss(debuff_damage)
+	playsound(debuff_owner.loc, "sound/bullets/acid_impact1.ogg", 5)
```
Indeed, that's why the volume is very low. It's intended to be spammed, but the super low volume counteracts it.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/sentinel/abilities_sentinel.dm:
```diff
@@ -1,36 +1,220 @@
+	ability_name = "toxic slash"
+	cooldown_timer = 10 SECONDS
+	plasma_cost = 100
+	use_state_flags = XACT_USE_BUCKLED
```
This is true of the target flag, but `XACT_USE_BUCKLED` allows Toxic Slash to be used while the Sentinel is riding a crusher, which was the intention of this change.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/sentinel/abilities_sentinel.dm:
```diff
@@ -1,36 +1,220 @@
+
+/datum/action/xeno_action/toxic_slash/action_activate()
+	. = ..()
+	var/mob/living/carbon/xenomorph/X = owner
```
Dunno which vars you're referring to. `toxic_slash_count` was renamed to `remaining_slashes` after this change as per the next review, and `toxic_slash_duration` seems proper enough, but I can rename it to `ability_duration` if that's somehow preferred.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/sentinel/abilities_sentinel.dm:
```diff
@@ -1,36 +1,220 @@
+///Called when Toxic Slash is active.
+/datum/action/xeno_action/toxic_slash/proc/toxic_slash(datum/source, mob/living/target, damage, list/damage_mod, list/armor_mod)
+	SIGNAL_HANDLER
+	var/mob/living/carbon/xenomorph/X = owner
```
Do you mean `carbon_target`? I can rename it to something else.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/sentinel/abilities_sentinel.dm:
```diff
@@ -1,36 +1,220 @@
+	carbon_target.balloon_alert(X, "Immune to Intoxication")
+
+///Called when Toxic Slash expires.
+/datum/action/xeno_action/toxic_slash/proc/toxic_slash_deactivate(mob/living/carbon/xenomorph/X)
```
I really gotta ask, what the heck is a proper name to you?

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/sentinel/abilities_sentinel.dm:
```diff
@@ -1,36 +1,220 @@
+		if(!silent)
+			owner.balloon_alert(owner, "Cannot sting")
+		return FALSE
+	var/mob/living/carbon/C = A
```
Ditto.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/sentinel/abilities_sentinel.dm:
```diff
@@ -1,36 +1,220 @@
+		return FALSE
+
+/datum/action/xeno_action/activable/drain_sting/use_ability(atom/A)
+	var/mob/living/carbon/xenomorph/X = owner
```
Ditto.

lKiseki commented on code/datums/status_effects/debuffs.dm:
```diff
@@ -418,3 +418,62 @@
+	particle_holder.particles.spawning = 1 + round(stacks / 2)
+	particle_holder.pixel_x = -2
+	particle_holder.pixel_y = 0
+	if(HAS_TRAIT(debuff_owner, TRAIT_INTOXICATION_RESISTANT))
```
Feels too complicated. I'll just make it so that a resistance makes it decay twice as fast.

lKiseki commented on code/modules/mob/living/carbon/xenomorph/castes/sentinel/abilities_sentinel.dm:
```diff
@@ -1,36 +1,220 @@
+	var/mob/living/carbon/C = A
+	var/datum/status_effect/stacking/intoxicated/debuff = C.has_status_effect(STATUS_EFFECT_INTOXICATED)
+	var/drain_potency = debuff.stacks * SENTINEL_DRAIN_MULTIPLIER
+	if(debuff.stacks > debuff.max_stacks - 10)
```
Already fixed. Resistance will make the debuff decay twice as fast instead.

lKiseki commented:
> Also just in terms of walance numbers, hard stun for up to 3 seconds seems rather strong, on top of the fat heal and other buffs. This seems like a bit of a 1v1 monster if played well.

Live testing has proven this to be relatively false, but I'll go ahead and nerf the knockdown duration a bit as a preventive measure.

lKiseki commented:
### Debuff: Intoxication
- Being resistant to the debuff will no longer lower the limit on stacks. Instead, it will make the debuff decay twice as fast.
- Slowdown and stagger granted by this debuff will only be applied over 20 stacks.
##
### Drain Sting
- Knockdown duration is now equal to 1 second for every 10 stacks, minus 1 second. For example, 30 stacks would result in a 2-second knockdown.

UntoldTactics commented:
Sentinels 100% need a buff to health and/or armor, ultra squishy for what the new set of abilities require

lKiseki commented:
### Bug Fixes
- Fixed a bug where the Intoxication debuff would runtime while trying to display a balloon alert.
- Fixed an oversight where Drain Sting could remove knockdown stacks if there were too few Intoxication stacks on the target.
##
### Caste Stats
- Movement speed will now scale with maturity, starting at -0.8 and ending at -0.9.
- Spit speed will now scale with maturity, starting at 1.3 seconds and ending at 1.0 seconds.
- The amount of stacks applied with abilities will now scale based on maturity, starting at 2 and ending at 5.
##
### Drain Sting
- Changed the visual effect that appears when the ability is used.
- Changed the visual effect that appears when the ability's critical effect is activated.
- Changed the visual effect for Drain Surge, the buff that is applied in the ability's critical effect.
- Victims will now scream when affected by the critical effect.

QualityVan commented on code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:
```diff
@@ -129,8 +129,10 @@
 					continue
 				else
 					neuro_applied = TRUE
-			defile_strength_multiplier *= 2
```
Did you remove this on purpose? It seems outside scope.

lKiseki replied:
Nope. Whoops.

MisterGrimm commented:
This redesign is pretty cool, but it honestly just isn't as good as the old neuro sentinel. The intoxication debuff does laughable damage, to the point that most marines don't even bother to resist. From my experience the marine just pops a pill and ignores the status, which almost always outheals the intoxication damage. Either the intoxication needs a higher damage ceiling, or the intoxication needs to have a interaction with medicine to prevent people from ignoring it.

TiviPlus reviewed (approved):
Fix CI failing due to icondiff error reading the dmi and this is fine
